### PR TITLE
OBPIH-5825 Add comments count on comments tab

### PR DESCRIPTION
**Pull request by drodzewicz**
- add comment count badge
- sort comments desc by last updated
- show comments only on requests (check by SourceType == Electronic)

### CODE REVIEW

File: grails-app/views/stockMovement/_comments.gsp [hunk 1]
```diff
@@ -12,7 +12,7 @@
             </tr>
             </thead>
             <tbody>
-            <g:each var="comment" in="${requisition?.comments}">
+            <g:each var="comment" in="${requisition?.comments?.sort(){ a,b -> b?.lastUpdated <=> a?.lastUpdated }}">
```
Comment (alannadolny) [inline grails-app/views/stockMovement/_comments.gsp:15]: Maybe we can create a method for comparing `lastUpdated` dates in the domain, and use the `.sort()` method without passing closure?

**Review by alannadolny — APPROVED** [review]

File: grails-app/views/stockMovement/show.gsp [hunk 2]
```diff
@@ -555,10 +556,14 @@
...
-                        <li>
+                    <g:if test="${stockMovement.sourceType == RequisitionSourceType.ELECTRONIC}">
+                        <g:set var="commentCount" value="${stockMovement?.requisition?.comments?.size()}" />
+                        <li
+                            data-count="${commentCount < 1000 ? commentCount : '999+' }"
+                            class="${commentCount > 0 ? 'tab-badge' : ''}"
+                        >
                             <g:link controller="stockMovement" action="comments" params="${[ id: stockMovement?.id ]}">
-                                <warehouse:message code="comments.label" default="Comments"/>
+                                <warehouse:message code="comments.label" default="Comments"/>${fragment}
```
Comment (kchelstowski) [inline grails-app/views/stockMovement/show.gsp:566]: what is `${fragment}`?

**Review by kchelstowski — APPROVED** [review]

File: grails-app/views/stockMovement/show.gsp (hunk 2 above)
Comment (drodzewicz) [reply grails-app/views/stockMovement/show.gsp:566]: I was testing how to extract # from the URL, but it didn't. Thanks for noticing that, it needs to be removed.

File: grails-app/views/stockMovement/_comments.gsp (hunk 1 above)
Comment (drodzewicz) [reply grails-app/views/stockMovement/_comments.gsp:15]: Just like Kacper suggested, I'll overwrite compareTo and call sort() in the GSP.

File: grails-app/views/stockMovement/show.gsp [hunk 3]
```diff
@@ -28,7 +29,7 @@
         <g:if test="${stockMovement?.documents}">
             <div class="right">
                 <div class="button-group">
-                    <g:if test="${stockMovement?.requisition}">
+                    <g:if test="${stockMovement.sourceType == RequisitionSourceType.ELECTRONIC}">
```
[the remark this answers is not in the document]
Comment (drodzewicz) [reply grails-app/views/stockMovement/show.gsp:32]: more like an oversight on my part

[the remark this answers is not in the document]
Comment (awalkowiak) [reply grails-app/views/stockMovement/show.gsp:32]: You can use `stockMovement.electronicType`

**Review by awalkowiak — COMMENTED** [review]
Beside what others noticed, it LGTM (I have a one tiny suggestion in the comments)

**Review by kchelstowski — APPROVED** [review]

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**